### PR TITLE
Misra: fix rule 20.10, and some random fixes

### PR DESCRIPTION
**Pull request by alfred2g**
Misra: fix rule 20.10, and some random fixes

Description
-----------
Misra: fix rule 20.10, and some random fixes such as adding U to some assignments and some variable name changes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by shubnil — APPROVED** [review]

File: source/FreeRTOS_Sockets.c [hunk 1]
```diff
@@ -4100,7 +4100,7 @@ void vSocketWakeUpUser( FreeRTOS_Socket_t * pxSocket )
         {
             size_t uxLittlePerc = sock20_PERCENT;
             size_t uxEnoughPerc = sock80_PERCENT;
-            size_t uxSegmentCount = pxSocket->u.xTCP.uxRxStreamSize / pxSocket->u.xTCP.usMSS;
+            size_t ulSegmentCount = pxSocket->u.xTCP.uxRxStreamSize / pxSocket->u.xTCP.usMSS;
```
Comment (paulbartell) [inline source/FreeRTOS_Sockets.c:4103]: size_t should have a ux prefix right?

File: source/portable/BufferManagement/BufferAllocation_2.c [hunk 2]
```diff
@@ -64,6 +64,7 @@
 #endif /* ipconfigUSE_TCP == 1 */
 
 /*_RB_ This is too complex not to have an explanation. */
+/* coverity[misra_c_2012_rule_20_10_violation] */
```
Comment (paulbartell) [inline source/portable/BufferManagement/BufferAllocation_2.c:67]: Consider making this a project wide exclusion.

**Review by paulbartell — CHANGES_REQUESTED** [review]
Changes in source/FreeRTOS_Sockets.c should be reverted or ux vs ul usage should be made consistent.

**Review by AniruddhaKanhere — APPROVED** [review]

File: source/portable/BufferManagement/BufferAllocation_2.c (hunk 2 above)
Comment (alfred2g) [reply source/portable/BufferManagement/BufferAllocation_2.c:67]: it would make similar macro additions go unnoticed

**Review by shubnil — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**